### ISSUE TITLE
access log is not written out with logback-access.xml

### ISSUE THREAD

**Issue by sridhar1982** [logstash/logstash-logback-encoder#180]
I am trying to use` LogstashAccessEncoder` to log request level fields (such as status code etc). Here is my configuration for `Logback.xml` and `Logback-access.xml`. when I run the service, I  do not get anything written 

logback.xml

```
<configuration>
    <property name="LOG_PATH" value="/tmp/json-log.json" />
    <appender name="jsonAppender" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <File>${LOG_PATH}</File>
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
        <includeCallerData>true</includeCallerData>
            <customFields>{"appname":"myWebservice"}</customFields>
        </encoder>
        <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
            <maxIndex>1</maxIndex>
            <fileNamePattern>${LOG_PATH}.%i</fileNamePattern>
        </rollingPolicy>
        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
            <MaxFileSize>1MB</MaxFileSize>
        </triggeringPolicy>
    </appender>
    <appender name="consoleAppender" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
            <customFields>{"appname":"myWebservice"}</customFields>
        <includeCallerData>true</includeCallerData>
        </encoder>
    </appender>
    <logger name="jsonLogger" additivity="false" level="DEBUG">
        <appender-ref ref="consoleAppender"/>
    </logger>
    <root level="INFO">
        <appender-ref ref="jsonAppender"/>
        <appender-ref ref="consoleAppender"/>
    </root>
</configuration>
```

logback-access.xml

```
<configuration>
    <appender name="consoleAppender1" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashAccessEncoder">
            <customFields>{"appname":"myWebservice"}</customFields>
            <fieldNames>
                <fieldsRequestHeaders>@fields.request_headers</fieldsRequestHeaders>
                <fieldsResponseHeaders>@fields.response_headers</fieldsResponseHeaders>
                <fieldsRequestedUrl>@fields.requested_url</fieldsRequestedUrl>
                <fieldsStatusCode>@fields.status_code</fieldsStatusCode>
            </fieldNames>
        </encoder>
    </appender>
    <root level="INFO">
        <appender-ref ref="consoleAppender1"/>
    </root>
</configuration>
```

Here is the output. As you can see, I don't see any access level information (such as status code, request url) in the message.

`{"@timestamp":"2016-08-09T13:11:48.397-07:00","@version":1,"message":"log message3 Greeting=id -> 1; content -> Hello, World!","logger_name":"in.sdqali.spring.GreetingController","thread_name":"http-nio-2020-exec-1","level":"INFO","level_value":20000,"HOSTNAME":"13MBP-13447.local","appname":"myWebservice","caller_class_name":"in.sdqali.spring.GreetingController","caller_method_name":"greeting","caller_file_name":"GreetingController.java","caller_line_number":33}`

Do I have to anything special to write to accessAppender?

This is the web request I make:
http://localhost:2020/greeting

**Comment by philsttr** [comment]
I think the problem is your `<root>` element in your `logback-access.xml` file.

I just tested the following file with jetty:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration debug="false" scan="false">
  <statusListener class="ch.qos.logback.core.status.OnConsoleStatusListener"/>

  <appender name="logstash-access" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashAccessEncoder">
            <customFields>{"appname":"myWebservice"}</customFields>
            <fieldNames>
                <fieldsRequestHeaders>@fields.request_headers</fieldsRequestHeaders>
                <fieldsResponseHeaders>@fields.response_headers</fieldsResponseHeaders>
                <fieldsRequestedUrl>@fields.requested_url</fieldsRequestedUrl>
                <fieldsStatusCode>@fields.status_code</fieldsStatusCode>
            </fieldNames>
        </encoder>
  </appender>

  <appender-ref ref="logstash-access"/>

</configuration>
```

And received the following output when I hit the root url:

```
{"@timestamp":"2016-08-12T23:08:46.414-07:00","@version":1,"@message":"127.0.0.1 - - [2016-08-12T23:08:46.414-07:00] \"GET / HTTP/1.1\" 304 0","@fields.method":"GET","@fields.protocol":"HTTP/1.1","@fields.status_code":304,"@fields.requested_url":"GET / HTTP/1.1","@fields.requested_uri":"/","@fields.remote_host":"127.0.0.1","@fields.HOSTNAME":"127.0.0.1","@fields.content_length":0,"@fields.elapsed_time":18,"@fields.request_headers":{"Accept":"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8","Accept-Encoding":"gzip, deflate","Accept-Language":"en-US,en;q=0.5","Connection":"keep-alive","Cookie":"visited=yes","Host":"localhost:8080","If-Modified-Since":"Tue, 10 Mar 2015 13:54:35 GMT","User-Agent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:28.0) Gecko/20100101 Firefox/28.0"},"HOSTNAME":"mint","appname":"myWebservice"}

```

**Comment by sridhar1982** [comment]
Thanks for looking into this issue. what is wrong with my `<root>`. besides that, do you have to configure anything in jetty to enable logback-access logging. I am using spring boot with tomcat and it appears some configuration is required. http://stackoverflow.com/questions/20574972/spring-boot-jetty-tomcat-embedded-access-log-configuration

**Comment by philsttr** [comment]
I just followed the [logback instructions](http://logback.qos.ch/access.html) for setting up logback access with jetty.  There are separate instructions for tomcat.  The steps probably vary with spring boot a little, but I think the same concepts apply.

In those instructions, there is no `root` element.  All of the examples use an `appender-ref` element, so I did the same thing.  I think `root` is only applicable to logback classic (not logback access).
